src/monotonic.c

```diff
@@ -0,0 +1,109 @@
+#include "monotonic.h"
+#include <stddef.h>
+#include <stdlib.h>
+
+#undef NDEBUG
+#include <assert.h>
+
+
+/* Note: monotonicInit() should be called once.  However, each version is
+ *       designed to support being called multiple times without error.  */
+const char * monotonicInit();
+
+#if defined(USE_PROCESSOR_CLOCK) && defined(__x86_64__) && defined(__linux__)
+    #include <regex.h>
```

Comment:
Since Oran also brought this up, I'd ask that #ifdefs do not change the indentation level of code. It's more in line with the remaining style of Redis and seems to be the common preference here.